classes/02class/aws-networking/README.md

```diff
@@ -0,0 +1,167 @@
+
+- `private` subnet
+  - Workloads like databases or identity management systems will run here
+  - Any service running in this subnet won't be accessible direct from public networks (like internet). To access applications 
+  - Services can go out to the internet through a nat gateway.
+  - to access a workload here, a jumpbox from the public subnet is neccesary
+
+- `public` subnet
+  - good fit for frontend webservers where clients will have the first point of contact. From the webserver you can then communicate with backend servers running on private subnets
+  - nat gateways (translate private IP's to public IP's)
+
+Very often a `secure` subnet will also be presented and is mainly used for databases. The secure subnet won't have neither IN or OUT access to internet and can only be accessed from workloads running in `private` subnet.
+
+Don't get me wrong here, those are just conventions.
+
+## Cloud Networking
+
+Most of today's cloud providers share the same concept of Virtual Private Cloud (VPC) for stablishing a logically isolated section of network within your account.
+
+A VPC can then be sliced into multiple networks called subnets where you will have control on IP address ranges, route tables and control security within these sections.
+
+### AWS Networking
+
+Subnets within a VPC on AWS environment are bound to only one Availability Zone (AZ). To achieve high availability (in case one AZ goes off) of a workload, we need to create the one public/private subnet per AZ, so 4 subnets in total for this example.
+
+This article will now move into a tutorial style with information needed for creating a fully working VPC with a few subnets. Explicit instructions/commands/where to go on AWS console can easily be found on the provider documentation. 
+
+As always, we recommend to play around in the console and familiarise yourself with it. There is not better way to learn:
+
+What we will be doing:
+
+- ip addressing design
+- build a VPC
+  - subnet public
+    - internet gateway
+  - subnet private
+    - network gateway
+  - routing table
+
+#### VPC
+
+The first thing is to determine what will be the network size for the whole VPC. AWS supports networks up to `65.536` IP's reffered as a `/16` network. 
+
+It's always ideal to think ahead how many devices your network will support and only reserve the right amount of addresses need with some extra for future expansions. Although we won't need this many IP's, for this example we will be using the `/16` network as it will simplify the big picture understanding.
+
+
+|field|value|
+|-|-|
+|Name tag|devopsacademy|
+|IPv4 CIDR Block|10.0.0.0/16|
+|IPv6|no|
+|Tenancy|default
+
+There is no cost for creating a VPC.
+
+#### Subnets
+
+Subnets can then be sliced to smaller chunks of ip addressing to isolate workloads. The most common size for a subnet is a `/24` which will support `256` IP's.
```

Comment:
`The VPC subnet can then be sliced into smaller subnets to spread the workloads between multiple AZs as well to isolate public and private workloads.` might be better